enhancements/ingress/operator-resource-configuration.md

```diff
@@ -0,0 +1,900 @@
+Burstable QoS and may be throttled or evicted under cluster resource pressure, 
+causing service disruptions.
+
+Acceptance Criteria:
+- Can configure resource limits matching requests for Guaranteed QoS
+- Router pods maintain stable performance under cluster resource pressure
+- Configuration survives router pod restarts and upgrades
+
+#### Story 3: Resource-Constrained Edge Deployment
+
+As an operator of an edge computing deployment with limited resources, I need to 
+set strict resource limits on ingress router pods to prevent them from consuming 
+excessive resources and impacting other critical workloads. Without limits, router 
+pods with Burstable QoS can burst beyond their requests, potentially starving other 
+pods in the resource-constrained environment.
+
+Acceptance Criteria:
+- Can configure resource limits to cap maximum resource consumption
+- Router pods do not exceed defined resource boundaries
+- Guaranteed QoS ensures router pods get their allocated resources under pressure
+- Other workloads on the node are protected from router resource overuse
+
+### Goals
+
+- Allow configuration of resource limits for ingress router pods (HAProxy containers)
+- Add feature to v1 API protected by a feature gate for simplicity
+- Maintain backward compatibility with existing IngressController v1 API
+- Use feature gate for Tech Preview period, then promote to Default feature set for GA
+- Enable router pods to achieve Guaranteed QoS class
+- Provide sensible defaults that work for most deployments
+- Support configuration for router container and sidecar containers (logs, metrics)
+
+### Non-Goals
+
+- Configuring resources for the ingress-operator deployment itself (the controller)
+- Auto-scaling or dynamic resource adjustment based on traffic load
+- Modifying the existing v1 API `spec.nodePlacement.resources` field (remains unchanged)
+- Creating a separate v1alpha1 API version (using v1 with feature gate instead)
+- Vertical Pod Autoscaler (VPA) integration (may be future work)
+- Horizontal Pod Autoscaler (HPA) configuration (separate concern)
+
+## Proposal
+
+### Workflow Description
+
+**Platform Administrator** is a human responsible for configuring the OpenShift cluster.
+
+1. Platform administrator enables the `IngressRouterResourceLimits` feature gate
+2. Platform administrator creates or updates an IngressController CR using the v1 API
+3. Platform administrator sets the new `resources` field with desired resource limits/requests
+4. The ingress-operator watches for changes to the IngressController CR
+5. The ingress-operator reconciles the router deployment with the specified resources
+6. Kubernetes performs a rolling restart of the router pods with the new resource configuration
+7. Router pods achieve Guaranteed QoS class (when limits == requests)
+8. Platform administrator verifies the changes with `oc describe deployment router-default -n openshift-ingress`
```

Comment:
Alternatively, we could have IngressControllerStatus also introduce a subfield for this to reflect if it properly applied, but maybe that's not needed given that it's relatively easy to view the deployment as well.